### PR TITLE
docker pr#16609 support: PullBehavior 

### PR DESCRIPTION
**Pull request by dkjer**
Changes to support docker https://github.com/docker/docker/pull/16609

Introduces the PullBehavior type.  See https://github.com/docker/docker/pull/16609 for more information

### CODE REVIEW

File: types/image/pull_behavior.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+		return PullNever, nil
+	case "always":
+		return PullAlways, nil
+	case "missing", "":
+		return PullMissing, nil
+	}
+	return PullNever, fmt.Errorf("Invalid pull behavior '%s'", pullVal)
+}
+
+func (p PullBehavior) String() string {
```
Comment (calavera) [inline types/image/pull_behavior.go]: this function needs a comment.

File: types/image/errors.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+// ErrImageDoesNotExist creates an error returned when no image can be found for a reference.
+func ErrImageDoesNotExist(refOrID string) error {
+	return errDoesNotExist{refOrID}
+}
+
+type errDoesNotExist struct {
+	RefOrID string
+}
+
+func (e errDoesNotExist) Error() string {
```
Comment (calavera) [inline types/image/errors.go]: this function needs a comment.

**Comment by calavera** [comment]
to be honest, I kinda like the errors where they were before.

**Comment by dkjer** [comment]
@calavera Yeah, its not completely necessary to refactor those errors messages, if its a blocker.  

This was an attempt to refactor two errors that serve the same purpose:
* [docker: daemon.ErrImageDoesNotExist](https://github.com/docker/docker/blob/master/daemon/daemon.go#L94)
* [engine-api: client.imageNotFoundError](https://github.com/docker/engine-api/blob/master/client/errors.go#L12)

However, the refactor dilutes the intent of the original pull request.  I have removed the error handling changes.

**Comment by calavera** [comment]
Awesome, thanks.

LGTM

**Comment by dkjer** [comment]
@calavera What are the next steps for getting this merged?

**Comment by calavera** [comment]
waiting for another @docker/engine-api-maintainers to LGTM

**Comment by dongluochen** [comment]
LGTM

**Comment by dongluochen** [comment]
Thanks @dkjer.

**Comment by dkjer** [comment]
@calavera @dongluochen Thanks!  Due to the api change in behavior, how should I go about updating hack/vendor.sh in https://github.com/docker/docker/pull/16609 ?  Should we create a v0.2.3 tag, and update hack/vendor.sh to match?

**Pull request merged**